src/GitTools.jl

```diff
@@ -68,20 +68,45 @@ end
 
 const GITHUB_REGEX =
     r"^(?:git@|git://|https://(?:[\w\.\+\-]+@)?)github.com[:/](([^/].+)/(.+?))(?:\.git)?$"i
-const GIT_PROTOCOL = Ref{Union{String, Nothing}}(nothing)
+const GITLAB_REGEX =
+    r"^(?:git@|git://|https://(?:[\w\.\+\-]+@)?)gitlab\.([^/:]+)[:/]((?:[^/].+)/(?:.+?))(?:\.git)?$"i
```

Comment:
Probably the approach should be to support any hostname. A more generalized regex can be seen [here](
https://github.com/JuliaLang/julia/blob/604046c1e1063d393261c07fc66a4f825b244697/stdlib/LibGit2/src/utils.jl#L9-L29).

Reply:
We don't need to go that with the regex though.